src/Listener/ItemSet.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylake\AkeneoProducerBundle\Listener;
+
+final class ItemSet implements ItemSetInterface
+{
+    /** @var object[] */
+    private $items = [];
+
+    /** {@inheritdoc} */
+    public function add($item)
+    {
+        if (!is_object($item)) {
+            throw new \InvalidArgumentException('Object storage accepts only objects!');
```

Comment:
An interface suggest that this should be `DomainException` instead of `InvalidArgumentException`